server/src/main/java/io/spine/server/BoundedContextBuilder.java

```diff
@@ -254,6 +218,27 @@ public BoundedContextBuilder addCommandDispatcher(CommandDispatcher<?> commandDi
         return this;
     }
 
+    /**
+     * Adds a filter for commands.
+     *
+     * <p>The order of appending the filters to the builder is the order of the filters in
+     * the {@code CommandBus}.
+     */
+    public BoundedContextBuilder addCommandFilter(BusFilter<CommandEnvelope> filter) {
+        checkNotNull(filter);
+        commandBus.addFilter(filter);
+        return this;
+    }
+
+    /**
+     * Adds a listener for commands posted to the {@code CommandBus} of the context being built.
+     */
+    public BoundedContextBuilder addCommandListener(Consumer<CommandEnvelope> listener) {
+        checkNotNull(listener);
+        commandBus.addListener(listener);
+        return this;
+    }
+
     /**
      * Adds the passed event dispatcher to the dispatcher registration list which will be processed
      * after the Bounded Context is created.
```

Comment:
The issue is located below, but I am not able to leave my comment there.

IDEA detects some code duplicates with this piece:
```java
        if (eventDispatcher instanceof Repository) {
            return add((Repository<?, ?>) eventDispatcher);
        }
        eventDispatchers.add(eventDispatcher);
        return this;
```